roles/awx/defaults/main.yml

```diff
@@ -68,15 +68,42 @@ tower_web_extra_volume_mounts: ''
 
 tower_redis_image: redis:latest
 
-tower_postgres_pass: awxpass
+ca_trust_bundle: "/etc/pki/tls/certs/ca-bundle.crt"
+
+development_mode: false
+
+
+# -- Database configuration
+#
+
+# Boolean to specify whether one wants the Operator to manage the database
+# deployment or want to rely on an externally managed database
+external_database: false
+
+# Container to deploy if user decides the operator should handle database
+# deployment
 tower_postgres_image: postgres:12
+
+# Persistent Volume Claim specification to customize for database data
 tower_postgres_storage_request: 8Gi
 tower_postgres_storage_class: ''
 
+# Mountpoint for database data
 tower_postgres_data_path: '/var/lib/postgresql/data/pgdata'
 
+# Custom name of the secret one wants the Operator to look into for connection
+# details. If not specified, the Operator will used the
+# tower_postgres_(user|database,...) variabes, if neither those are specified
+# the operator will look for <awx-instance-name>-postgres-configuration secret
+tower_postgres_configuration_secret: ''
+
+# In an external_database=true scenario, those are the variables used to
+# connect to the database - Not recommended. A secret should be used.
+#
+# In an external_database=false scenario, those are the variables used to
+# configure the database
+tower_postgres_user: awx
+tower_postgres_pass: ''
```

Comment:
Having these variables live in this file makes it seem like they can be provided directly by the user via a CR, which they can't / shouldn't be. They should only ever be looked up via a Secret.

Reply:
Well, this is actually the whole point of this PR. It is to be more flexible to the way a user can provide those data.

1. Provide a secret (name it the way you want) specify it as `tower_postgres_configuration_secret`
2. Provide connection data (user, pass, database, ...) with one variable per value (not recommended) but it allows an easy onboarding (think UI at the OLM level)
3. Default to the Operator to read `<instance-name>-postgres-configuration`

Reply:
Well, sorry, this isn't going to work 😅

Reply:
https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#should-i-use-a-configmap-or-a-custom-resource

>Note: Use a secret for sensitive data, which is similar to a configMap but more secure.